go/docs/github.go

```diff
@@ -0,0 +1,237 @@
+
+	return booklit.Styled{
+		Style:   "github-release-link",
+		Content: booklit.String(tag),
+		Partials: booklit.Partials{
+			"Owner": booklit.String("concourse"),
+			"Repo":  booklit.String(repo),
+		},
+	}
+}
+
+func (p Plugin) Ghpr(number string, optionalRepo ...string) booklit.Content {
+	repo := "concourse"
+	if len(optionalRepo) > 0 {
+		repo = optionalRepo[0]
+	}
+
+	return booklit.Styled{
+		Style:   "github-pr-link",
+		Content: booklit.String(number),
+		Partials: booklit.Partials{
+			"Owner": booklit.String("concourse"),
+			"Repo":  booklit.String(repo),
+		},
+	}
+}
+
+func (p Plugin) Ghissue(number string, optionalRepo ...string) booklit.Content {
+	repo := "concourse"
+	if len(optionalRepo) > 0 {
+		repo = optionalRepo[0]
+	}
+
+	return booklit.Styled{
+		Style:   "github-issue-link",
+		Content: booklit.String(number),
+		Partials: booklit.Partials{
+			"Owner": booklit.String("concourse"),
+			"Repo":  booklit.String(repo),
+		},
+	}
+}
+
+func (p Plugin) Ghlabel(name booklit.Content, colorHex string) (booklit.Content, error) {
+	color, err := colorful.Hex(colorHex)
+	if err != nil {
+		return nil, err
+	}
+
+	lightContrastRating := contrast(color, white)
+	darkContrastRating := contrast(color, color.BlendRgb(black, 0.8))
+
+	class := "dark-label"
+	if lightContrastRating > darkContrastRating {
+		class = "light-label"
+	}
+
+	return booklit.Styled{
+		Style:   "label",
+		Content: name,
+		Partials: booklit.Partials{
+			"Color": booklit.String(colorHex),
+			"Class": booklit.String(class),
+		},
+	}, nil
+}
+
+func (p *Plugin) DownloadLinks() (booklit.Content, error) {
+	ctx := context.Background()
+
+	client, ok := p.githubClient(ctx)
+	if !ok {
+		return fillerDownloads, nil
+	}
+
+	type GitHubRelease struct {
+		Name         string
+		TagName      *string
+		IsDraft      bool
+		IsPrerelease bool
+		Url          string
+		Assets       struct {
+			Nodes []struct {
+				Name        string
+				DownloadUrl string
+			}
+		} `graphql:"releaseAssets(first: 20)"`
+	}
+
+	var releasesQuery struct {
+		Repository struct {
+			Releases struct {
+				Nodes []GitHubRelease
+			} `graphql:"releases(first: 10, orderBy: {field: CREATED_AT, direction: DESC})"`
+		} `graphql:"repository(owner: \"concourse\", name: \"concourse\")"`
+	}
+
+	err := client.Query(ctx, &releasesQuery, nil)
+	if err != nil {
+		return nil, fmt.Errorf("fetch releases: %w", err)
+	}
+
+	releases := releasesQuery.Repository.Releases.Nodes
+
+	var latestRelease *GitHubRelease
+	var latestVersion semver.Version
+	if len(releases) == 0 {
+		return nil, errors.New("no releases found!")
+	}
+	for _, release := range releases {
+		if release.IsPrerelease {
+			continue
+		}
+
+		if release.TagName == nil {
+			return nil, fmt.Errorf("no tag name for release %v", release)
+		}
+		version, err := semver.ParseTolerant(*release.TagName)
+		if err != nil {
+			return nil, err
+		}
+
+		if latestRelease == nil || version.GT(latestVersion) {
+			latestRelease = &release
```

Comment:
I kept getting 6.5.1 as my latest version. You're returning a pointer to `release` which gets updated after every loop.
<img width="429" alt="image" src="https://user-images.githubusercontent.com/4583837/112052298-57546200-8b29-11eb-94ac-2ba0bc31286c.png">

Reply:
fixed!